lib/transaction/index.js

```diff
@@ -1314,4 +1318,8 @@ function addRequestParameters(requestParameters) {
   }
 }
 
+Transaction.prototype.addLogsToAggregator = function addLogsToAggregator() {
+  this.agent.logs.addBatch(this.logs, this.priority)
```

Comment:
Would recommend we clear out the collection on transaction end to free up memory as segments continue to get propagated.

Reply:
We will want an upper bound here... perhaps same as the total upper bound. Need to determine if dropping past the threshold or also doing a priority sampling per transaction.